src/pyhf/contrib/utils.py

```diff
@@ -62,17 +65,52 @@ def download(archive_url, output_directory, force=False, compress=False):
         # The HEPData landing page for the resource file can check if the Accept
         # request HTTP header matches the content type of the resource file and
         # return the content directly if so.
+        # TODO: Figure out how to accept both x-tar and zip headers. :?
+        # with requests.get(
+        #     archive_url, headers={"Accept": "application/zip"}
+        # ) as response:
```

Comment:
@kratsg You have any experience with this?

Reply:
Actually `.zip` seems to work with `application/x-tar`(?) so maybe this isn't needed?

Reply:
Sorry I didn't comment before you merged.  The `Accept` headers will be ignored if `archive_url` is a HEPData URL (ending in `?view=true`) that returns the content directly.  The `Accept` headers will only become relevant when `archive_url` is a HEPData DOI for a resource file after we redirect to a landing page.  We're still working on the relevant PR (HEPData/hepdata#412) after it expanded in scope.  The default `Accept` header is `*/*` for Python `requests` (and `curl`), different from the [default `Accept` values of most web browsers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation/List_of_default_Accept_values) where `*/*` has a lower weighting ([quality value](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values)) than `text/html`.  With our current implementation, if `*/*` has weight 1 in a request to the landing page, the content will be returned directly instead of the HTML landing page.  This means you don't strictly need to specify the `Accept` header with Python `requests`.  But I'd recommend you keep the `Accept` header to avoid accepting content types that your download code can't process.  To accept both tar and zip files, you just need to use `headers={"Accept": "application/x-tar, application/zip"}`.  We will return a [406 Not Acceptable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) status code if a request is made to the landing page where the `Accept` header does not match the media type of the content.

Reply:
Thanks @GraemeWatt! This is all super helpful and I truly appreciate the level of detail &mdash; helps a lot for someone like me where this isn't my forte.